pkg/webhooks/namespace/namespace.go

```diff
@@ -150,8 +152,29 @@ func (s *NamespaceWebhook) authorized(request admissionctl.Request) admissionctl
 			ret.UID = request.AdmissionRequest.UID
 			return ret
 		}
-		log.Info("Non-admin access attempt to privileged namespace", "request", request.AdmissionRequest)
-		ret = admissionctl.Denied(fmt.Sprintf("Prevented from creating or changing a potentially harmful Namespace. Customer namespaces should NOT match this regular expressions: %s", privilegedNamespace))
+		log.Info("Non-admin (not 'kube:admin' or 'system:admin', or a member of osd-sre-admins or osd-sre-cluster-admins) attempted to access a privileged namespace (eg matching this regex", "regex", privilegedNamespace, "request", request.AdmissionRequest)
+		ret = admissionctl.Denied(fmt.Sprintf("Prevented from accessing Red Hat managed namespaces. Customer workloads should be placed in customer namespaces, not 'default' or any namespace beginning with 'kube', 'openshift', or 'redhat'."))
+		ret.UID = request.AdmissionRequest.UID
+		return ret
+	}
+
+	if badNamespaceRe.Match([]byte(ns.GetName())) {
+		amISREAdmin := false
+		amIClusterAdmin := utils.SliceContains(request.UserInfo.Username, clusterAdminUsers)
+
+		for _, group := range sreAdminGroups {
+			if utils.SliceContains(group, request.UserInfo.Groups) {
+				amISREAdmin = true
+				break
+			}
+		}
+		if amIClusterAdmin || amISREAdmin {
+			ret = admissionctl.Allowed("Cluster and SRE admins may access")
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+		log.Info("Non-admin (not 'kube:admin' or 'system:admin', or a member of osd-sre-admins or osd-sre-cluster-admins) attempted to access a potentially harmful namespace (eg matching this regex", "regex", badNamespace, "request", request.AdmissionRequest)
+		ret = admissionctl.Denied(fmt.Sprintf("Prevented from creating a potentially harmful namespace. Customer namespaces should not begin with 'com', 'io', or 'in' as this would impact DNS resolution."))
```

Comment:
With the log line I think we can omit the list of groups here because we'll know what code is in use and what safelisted groups are, so it's extra stuff.

> Customer namespaces should not begin with 'com', 'io', or 'in' as this would impact DNS resolution.

Check the regular expression :) It's `^com$` (etc) which means the name can't be exactly `com` (etc). May want to provide the regex here as well, like with the previous check.

Reply:
+1. fixed again